pkg/client/tx/client.go

```diff
@@ -404,6 +376,36 @@ func (tClient *txClient) goSubscribeToOwnTxs(ctx context.Context) {
 			if len(txErrorChans) == 0 {
 				delete(tClient.txTimeoutPool, timeoutHeight)
 			}
+
+			// Check for a corresponding error channel in the map.
+			// Only close and remove from tClient.txErrorChans if it has been initiated
+			// by tClient.
+			// Transactions that are initiated externally will not have an associated
+			// error channel in tClient.txErrorChans and would trigger a panic if not
+			// handled inside this loop.
+			txErrCh, ok := tClient.txErrorChans[txHashHex]
+			if !ok {
+				panic("Received tx event without an associated error channel.")
+			}
+
+			// TODO_INVESTIGATE: it seems like it may not be possible for the
```

Comment:
Can you please shorten & simplify this. Reading the comment made it unclear to me whether our code path is callining `Sync` or `Async` broadcasting.

Reply:
Updated the comment. @bryanchriswhite, could you please check if the shortened comment is still a valid?